archiver.rb

```diff
@@ -0,0 +1,26 @@
+class Archiver
+  attr_accessor :file
+  @@modes = [:reader, :writer, :both]
+
+  @@modes.each do |mode|
+    define_method("open_file_as_#{mode}") do |file|
+      @file = File.open(file, 'r') if mode == :reader
+      @file = File.open(file, 'w') if mode == :writer
+      @file = File.open(file, 'r+') if mode == :both
+    end
+  end
+
+  def read_file lines: false
+    lines ? data = self.file&.read : data = self.file&.readlines.map(&:chomp)
```

Comment:
It is not necessary to declare `data` variable. Also self is not necessary in the scope context `lines ? file&.read : data = file&.readlines.map(&:chomp)`